### ISSUE TITLE
Severe performance issues with PB on Valiant Vaccinated Edition MAP07

### ISSUE THREAD

**Issue by Valmar33** [pa1nki113r/Project_Brutality#672]
Performance is extremely bad.

No perf issues if I load the map without PB.

**Comment by JMartinez9820** [comment]
My advice is to use LZDoom or GZDoom GLES. To be honest, Valiant is very unoptimized, and running PB doesn't make it better. Performance improvements will come way later in development, and I believe this may be a duplicate of several other issues too

**Comment by Valmar33** [comment]
> My advice is to use LZDoom or GZDoom GLES. To be honest, Valiant is very unoptimized, and running PB doesn't make it better. Performance improvements will come way later in development, and I believe this may be a duplicate of several other issues too

I completely forgot to mention that I'm running it with GZDoom master with default config on Linux :eyes:

Issue remains whether I choose OpenGL or Vulkan backends.

**Comment by Cacodemon345** [comment]
Does it run faster with Release/RelWithDebInfo builds?

**Comment by Valmar33** [comment]
> Does it run faster with Release/RelWithDebInfo builds of GZDoom?

Already running it very close to a "Release" config, I think?

Build settings:

```
cmake -B build \
          -G Ninja \
          -D CMAKE_BUILD_TYPE="None" \
          -D CMAKE_C_FLAGS="${CFLAGS} -ffile-prefix-map=\"$PWD\"=. -DSHARE_DIR=\\\"/usr/share/gzdoom\\\"" \
          -D CMAKE_CXX_FLAGS="${CXXFLAGS} -ffile-prefix-map=\"$PWD\"=. -DSHARE_DIR=\\\"/usr/share/gzdoom\\\"" \
          -D CMAKE_EXE_LINKER_FLAGS="${LDFLAGS}" \
          -D DYN_GTK=OFF \
          -D DYN_OPENAL=OFF
```

**Comment by popguy12** [comment]
so, the barrels in PB have an ACS called specifically for Valiant, but it seems to be for Map18, perhaps that might be something to look into

**Comment by BLOODWOLF333** [comment]
yeah the ExplosiveBarrel1 actor is causing all this lag.  It will also drag the game to a crawl on Sunlust map30

**Comment by popguy12** [comment]
apparently, this entire issue has to do with a crusher thats crushing barrels, causing immense amounts of lag
PA1N originally coded in that valiant fix barrel, but i dont think its really gonna be needed once the barrel overhaul is done

**Comment by JMartinez9820** [comment]
@Valmar33
Here is a small patch/add-on that will fix the performance issues in Valiant VE Map 07. This also applies to Valiant's Map 18, since I think performance issues happen there too. This simple replaces the current PB Barrels with the Vanilla Doom barrels (thanks ZDoom Wiki for the code)

https://www.mediafire.com/file/toq62cumqan5f1c/PB_ValiantBarrelPerformanceFix.pk3/file
